File: src/1685.cpp
#include <vector>
#include <numeric>

using namespace std;

class Solution {
public:
    vector<int> getSumAbsoluteDifferences(vector<int>& nums) {
        vector<int> ret;
        int left = 0;
        int right = accumulate(nums.begin(), nums.end(), 0);
        for (int i = 0; i < nums.size(); ++i) {
            ret.push_back(i * nums[i] - left + right - (nums.size() - i) * nums[i]);
            left += nums[i];
            right -= nums[i];
        }
        return ret;
    }
};

File: src/1685.py
class Solution:
    def getSumAbsoluteDifferences(self, nums: list[int]) -> list[int]:
        ret = []
        left = 0
        right = sum(nums)
        for i in range(len(nums)):
            ret.append(i * nums[i] - left + right - (len(nums) - i) * nums[i])
            left += nums[i]
            right -= nums[i]
        return ret
